contracts/facets/ThirdPartyFacet.sol

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../libraries/LibTypes.sol";
+import "../libraries/LibEvents.sol";
+import {LibErrors} from "../libraries/LibErrors.sol";
+import {LibBonds} from "../libraries/LibBonds.sol";
+import {LibUtils} from "../libraries/LibUtils.sol";
+import {AppStorage} from "../storage/LibAppStorage.sol";
+
+contract ThirdPartyFacet {
+    AppStorage internal s;
+
+    /// @notice Close a sarcophagus that has not been unwrapped before its resurrection window is passed
+    /// @param identifier The sarcophagus ID
+    /// @param paymentAddress The address to which rewards will be sent
+    function clean(bytes32 identifier, address paymentAddress) external {
+        LibTypes.Sarcophagus storage sarco = s.sarcophaguses[identifier];
+
+        if (sarco.state != LibTypes.SarcophagusState.Exists) {
+            revert LibErrors.SarcophagusDoesNotExist(identifier);
+        }
+
+        // Make sure the sarco is cleanable
+        if (
+            block.timestamp <
+            LibUtils.getGracePeriod(sarco.resurrectionTime) +
+                sarco.resurrectionTime
+        ) {
+            revert LibErrors.SarcophagusNotCleanable();
+        }
+
+        // Figure out which archaeoligists did not fulfil their duties;
+        // accumulate their digging fees and bounties
+        address[] memory archAddresses = sarco.archaeologists;
+
+        uint256 totalCursedBond;
+        uint256 totalDiggingFee;
+        uint256 totalBounty;
+
+        for (uint256 i = 0; i < archAddresses.length; i++) {
+            if (!s.archaeologistSuccesses[archAddresses[i]][identifier]) {
```

Comment:
In a PR that hasn't been merged yet I added the unencrypted shards to the `archaeologistStorage` struct. Checking if an archaeologist has uploaded an unencrypted shard is another way to check if an archaeologist has done their job. 

We could create a function that pulls the unencrypted shard like this
```
bytes memory unencryptedShard = s.sarcophagusArchaeologists[identifier][archAddresses[i]].unencryptedShard;
```
And then checks that it's length is greater than 0.

Just leaving this note here so that we can consider changing it after everything else is merged.

Reply:
That seems like a better way to do this, yep. Maybe create a task for after the PR is merged?